sdk/bls.go

```diff
@@ -0,0 +1,156 @@
+package sdk
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	blst "github.com/supranational/blst/bindings/go"
+	"io"
+	"strings"
+)
+
+const FILECOIN_DST = string("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_")
+const ETH2_DST = string("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+
+// It must be a 32 byte sequence.
+type BLSPrivateKey string
+
+var ZeroBLSPrivateKey = BLSPrivateKey(make([]byte, 32))
+
+// It must be a 48 byte sequence.
+type BLSPublicKey string
+
+var ZeroBLSPublicKey = BLSPublicKey(make([]byte, 48))
+
+// It must be a 96 byte sequence.
+type BLSSignature string
+
+var ZeroBLSSignature = BLSSignature(make([]byte, 96))
+
+func GeneratePrivateKey(seed io.Reader) BLSPrivateKey {
+	if seed == nil {
```

Comment:
just want to clarify if we should handle `nil` interfaces that actually are not `nil`? For example, if I make something like this
```
var zr *ZeroReader
GeneratePrivateKey(zr)
```

i'll catch nil pointer panic. In this case, `zr` is `nil`, but `seed` actually is not `nil`